extra/nrnivmodl-core.in

```diff
@@ -1,104 +1,121 @@
 #!/bin/bash
+
 # -----------------------------------------------------------------------------
-# nrnivmodl-core uses make to compile additonal mod files for coreneuron
-# Copyright 2019 Blue Brain Project
+# Copyright 2020 Blue Brain Project
+# nrnivmodl-core is used to compile mod files for coreneuron
 # -----------------------------------------------------------------------------
+
 set -e
 
+# name of the script
 APP_NAME=$(basename $0)
-_PARENT="$(dirname $BASH_SOURCE)/.."
-ROOTDIR=$(perl -e "use Cwd 'abs_path'; print abs_path('$_PARENT')")
-MAKE_OPTIONS="MECH_NAME NMODL_BINARY DESTDIR INCFLAGS LINKFLAGS MODS_PATH VERBOSE"
-PARALLELISM=4
-SAVE_FILE="@CMAKE_HOST_SYSTEM_PROCESSOR@/nrnivmodl_options.txt"
+
+# directory and parent directory of this script
+PARENT_DIR="$(dirname $BASH_SOURCE)/.."
+ROOT_DIR=$(@PERL_EXECUTABLE@ -e "use Cwd 'abs_path'; print abs_path('$PARENT_DIR')")
+
+# default arguments : number of parallel builds and default mod file path
+PARALLEL_BUILDS=4
 params_MODS_PATH="."
 
-# To support build + install later we save/restore options
-# Attempt to load previous definitions
-if [ -f "$SAVE_FILE" ]; then
-    echo "[INFO] Loading definitions from previous nrnivmodl-core execution"
-    while read line; do
-        echo " + $line" && eval "$line"
-    done < "$SAVE_FILE"
-fi
+# prefix for common options : make sure to rename these if options are changed.
+MAKE_OPTIONS="MECHLIB_SUFFIX NMODL_BINARY INCFLAGS LINKFLAGS MODS_PATH VERBOSE"
 
-# Parse args (overriding)
+# parse CLI args
 while getopts "n:m:v:d:i:l:p:hV" OPT; do
     case "$OPT" in
     n)
-        params_MECH_NAME="$OPTARG";;
+        # suffix for mechanism library
+        params_MECHLIB_SUFFIX="$OPTARG";;
     m)
+        # nmodl or mod2c binary to use
         params_NMODL_BINARY="$OPTARG";;
-    d)
```

Comment:
This option allowed us to separate build from install steps, which is used in the Spack recipe. It might add some code but I believe it's a good concept to only start installing when the build succeeded, while avoiding to rebuild again.

Reply:
On popular demand, I re-introduced `-d` option and `install` target :smiley: